test/action_controller/serialization_test.rb

```diff
@@ -378,6 +375,45 @@ def test_cache_expiration_on_update
         assert_equal 'application/json', @response.content_type
         assert_equal expected.to_json, @response.body
       end
+
+      def test_fail_calling_use_adapter_on_controller_instance
+        controller = ImplicitSerializationTestController.new
+        assert_raises(ActiveModel::Serializer::Error) {
+          controller.use_adapter?
+        }
+      end
+
+      # FIXME: Test is currently failing
+      def test_fail_calling_use_adapter_on_controller_that_overrides_use_adapter
+        controller = Class.new(ImplicitSerializationTestController) {
+          def use_adapter?
+            false
+          end
+        }.new
+        assert_raises(ActiveModel::Serializer::Error) {
+          controller.use_adapter?
+        }
+      end
```

Comment:
The two failing controller tests are because the 'deprecated' controller methods that raise the error are being overridden, and hence not called.  I'm not sure how to handle that, so adding this failing test to call attention to it.